api/v1alpha1/storageclaim_types.go

```diff
@@ -52,10 +54,10 @@ type StorageClientNamespacedName struct {
 // StorageClaimSpec defines the desired state of StorageClaim
 type StorageClaimSpec struct {
 	//+kubebuilder:validation:Enum=block;sharedfile
-	Type             string                       `json:"type"`
-	EncryptionMethod string                       `json:"encryptionMethod,omitempty"`
-	StorageProfile   string                       `json:"storageProfile,omitempty"`
-	StorageClient    *StorageClientNamespacedName `json:"storageClient"`
+	Type              string  `json:"type"`
+	EncryptionMethod  string  `json:"encryptionMethod,omitempty"`
+	StorageProfile    string  `json:"storageProfile,omitempty"`
+	StorageClientName *string `json:"storageClient"`
```

Comment:
Why the Name prefix? I understand we are changing the type. buy why do we need to change the name of the field?

Reply:
No legit reason, only explicit mention.

Reply:
Let's remove it. This is part of a k8s API we need to be consistent. 
(e.g we have `StoageProfile` and not  `StoageProfileName`, etc.)

Reply:
done.